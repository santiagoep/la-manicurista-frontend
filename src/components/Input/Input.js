import { StyledContainer, StyledInput } from './Input.styled';

const Input = (props) => (
  <StyledContainer>
    <StyledInput {...props} type='text' />
  </StyledContainer>
);

export default Input;
