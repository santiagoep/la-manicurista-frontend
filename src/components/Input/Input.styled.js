import styled from 'styled-components';

export const StyledContainer = styled.div`
  display: flex;
  flex-grow: 1;
  flex-shrink: 0;
  position: relative;
  background-color: transparent;
  padding: ${({ theme }) => theme.utils.spacing(1, 4)};
  &::after {
    content: '';
    position: absolute;
    left: 0;
    right: 0;
    bottom: 0;
    top: 54.84%;
    border: 1px solid ${({ theme }) => theme.colors.borders.primary};
    box-sizing: border-box;
    border-radius: 0 0 20px 20px;
    border-top-style: none;
  }
`;

export const StyledInput = styled.input`
  width: calc(100% - ${({ theme }) => theme.utils.spacing(8)});
  border: none;
  outline: none;
`;
