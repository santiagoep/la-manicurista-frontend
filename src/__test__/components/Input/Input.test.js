import React from 'react';

import render from '../../render';
import Input from '@components/Input/Input';

describe('<Input />', () => {
  it('should render snapshot', () => {
    const { renderer } = render({ theme: true, redux: true });
    const component = renderer(<Input value='Hola' />);
    expect(component).toMatchSnapshot();
  });
  it('should onChange', () => {
    const spy = jest.fn();
    const { mount } = render({ theme: true, redux: true });
    const component = mount(<Input placeholder='test' onChange={spy} />);
    component.find('input').simulate('change', 'Test');
    expect(spy).toHaveBeenCalled();
  });
});
