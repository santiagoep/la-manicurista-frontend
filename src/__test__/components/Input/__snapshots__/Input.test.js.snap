// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render snapshot 1`] = `
<div
  className="Inputstyled__StyledContainer-sc-792yv1-0 rXHmE"
>
  <input
    className="Inputstyled__StyledInput-sc-792yv1-1 bwNuSi"
    type="text"
    value="Hola"
  />
</div>
`;
